rajawali/src/main/java/org/rajawali3d/loader/LoaderOBJ.java

```diff
@@ -421,7 +422,7 @@ private static String generateObjectName() {
 
 	private static String generateObjectName(String prefix) {
 		if(prefix==null) prefix = "Object";
-		return prefix + (int) (Math.random() * 9e3 + 1e3);
+		return prefix + Long.toString(SystemClock.elapsedRealtime());
```

Comment:
Keep in mind, this is in nanos, I would go with millis `System.currentTimeMillis()` which is more common to use as unixtimestamp. But I'm fine with it.

Reply:
`System.currentTimeMillis()` would not necessarily provide the necessary resolution, we can parse multiple child objects in a mS

Reply:
I see you guys have now implemented this feature. Is there away to pull these changes or can it be released? I'm working on something really urgent and i would love to have this feature. Thank you!

Reply:
Really urgent ? Nothing to do, here you go 
`implementation "com.github.contriteobserver:Rajawali:LoadedMtlObjNaming-SNAPSHOT"`
or
`implementation "com.github.contriteobserver:Rajawali:45cf99dc15"`

With jitpack.io you can have a library of any commit. Please see [details](https://jitpack.io/#contriteobserver/Rajawali/45cf99dc15)

Reply:
Yes, definitely urgent! thank you again!

Reply:
If the object names need guaranteed uniqueness, why not a uuid?

Reply:
UUIDs alone are insufficient, as we need the name to be revealing. Also, the legacy identifiers had the classifier "Object" and the instance label "[0-9][0-9][0-9][0-9]" select-able by character type (letters vs. numbers) it seemed appropriate to preserve that ability. 

Also, I an uncertain as to the overhead of a UUID call, as we need to interface with the MAC address, the clock, pseudo-random generator, and hashing. Was worried a UUID call might slow down file parsing.

Reply:
Maybe we should consider using an atomic number generator then. Maybe useful as the object IDs should be consistent as long as the model or loading order does not change.

Reply:
Implemented an `AtomicInteger` based identification mechanism, the `PosingDoll` object graph now looks like:

 I/MaterialNamingRenderer.InitScreen: object graph
    -->GROUP default
    	-->Arm.Left1002
    	-->Arm.Right1003
    	-->Leg.Left1004
    	-->Leg.Right1005
    	-->Head1006
    	-->Standing_Pose